github-repo-manager/src/main/java/com/axonivy/github/file/GitHubMissingFilesDetector.java

```diff
@@ -0,0 +1,118 @@
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHRepository;
+
+import com.axonivy.github.DryRun;
+import com.axonivy.github.GitHubProvider;
+
+public class GitHubMissingFilesDetector {
+
+  private static final String GITHUB_ORG = ".github";
+  private final byte[] fileContent;
+  private boolean isMissingRequiredFile;
+  private List<String> workingRepos;
+  private FileMeta fileMeta;
+
+  public GitHubMissingFilesDetector(FileMeta fileMeta, List<String> workingRepos) throws IOException {
+    Objects.requireNonNull(fileMeta);
+    Objects.requireNonNull(workingRepos);
+    this.fileMeta = fileMeta;
+    this.workingRepos = workingRepos;
+    try (var is = GitHubMissingFilesDetector.class.getResourceAsStream(fileMeta.requiredFilePath)) {
+      if (is == null) {
+        throw new IOException(fileMeta.requiredFilePath + " file not found");
+      }
+      this.fileContent = IOUtils.toByteArray(is);
+    }
+  }
+
+  public int checkMissingFile() throws IOException {
+    var github = GitHubProvider.get();
+    printInfoMessage("Working on organizations: {0}.", workingRepos);
+    for (var orgName : workingRepos) {
+      var org = github.getOrganization(orgName);
+      for (var repo : List.copyOf(org.getRepositories().values())) {
+        checkMissingFile(repo);
+      }
+    }
+    if (isMissingRequiredFile) {
+      printErrorMessage("At least one repository has no {0}.", fileMeta.requiredFilePath);
+      printErrorMessage("Add a {0} manually or run the build without DRYRUN to add {0} to the repository.",
+          fileMeta.requiredFilePath);
+      return -1;
+    }
+    return 0;
+  }
+
+  private void checkMissingFile(GHRepository repo) throws IOException {
+    if (repo.isPrivate() || repo.isArchived()) {
+      printInfoMessage("Repo {0} is {1}.", repo.getFullName(), repo.isPrivate() ? "private" : "archived");
+      return;
+    }
+
+    var foundFile = getFileContent(fileMeta.requiredFilePath, repo);
+    if (foundFile != null) {
+      if (!hasSimilarFilePaths(foundFile)) {
+        printInfoMessage("Repo {0} has {1} but no {2} file instead it has {3}.", repo.getFullName(),
+            foundFile.getName(), fileMeta.requiredFilePath, foundFile.getPath());
+        isMissingRequiredFile = true;
+      } else {
+        printInfoMessage("Repo {0} has {1}.", repo.getFullName(), foundFile.getName());
+      }
+    } else if (!GITHUB_ORG.equals(repo.getName())) {
+      addMissingFile(repo);
+    }
+  }
+
+  private GHContent getFileContent(String path, GHRepository repo) {
+    try {
+      return repo.getFileContent(path);
+    } catch (Exception e) {
+      printErrorMessage("File {0} in repo {1} is not found.", path, repo.getFullName());
+      return null;
+    }
+  }
+
+  private boolean hasSimilarFilePaths(GHContent existingFile) {
```

Comment:
This method does not make sense anymore. It was here because getLicense() returned licenses in LICENSE and COPYING file. Since you now work with getFileContent() this check no longer makes sense.

Instead, we may want to check if the content of the file is the same as the standard one?